numba/cuda/tests/cudapy/test_intrinsics.py

```diff
@@ -445,6 +466,72 @@ def test_hmul_ptx(self):
         ptx, _ = compile_ptx(simple_hmul_scalar, args, cc=(5, 3))
         self.assertIn('mul.f16', ptx)
 
+    @skip_unless_cc_53
+    def test_hdiv_scalar(self):
+        compiled = cuda.jit("void(f2[:], f2, f2)")(simple_hdiv_scalar)
+        ary = np.zeros(1, dtype=np.float16)
+        arg1 = np.float16(3.1415926)
+        arg2 = np.float16(1.57)
+        compiled[1, 1](ary, arg1, arg2)
+        ref = arg1 / arg2
+        np.testing.assert_allclose(ary[0], ref)
+
+    @skip_unless_cc_53
+    def test_hdiv(self):
+        compiled = cuda.jit("void(f2[:], f2[:], f2[:])")(simple_hdiv_kernel)
+        arry1 = np.random.randint(-65504, 65505, size=500).astype(np.float16)
+        arry2 = np.random.randint(-65504, 65505, size=500).astype(np.float16)
+        ary = np.zeros_like(arry1, dtype=np.float16)
+
+        compiled.forall(ary.size)(ary, arry1, arry2)
+        ref = arry1 / arry2
+        np.testing.assert_allclose(ary, ref)
+
+    @skip_unless_cc_53
+    def test_hdiv_denormal(self):
+        compiled = cuda.jit("void(f2[:], f2[:], f2[:])")(simple_hdiv_kernel)
+
+        # This test is designed to test divison of fp16 constants that
+        # result in small denormal numbers that should not be further
+        # iterated using the Newton Raphson algorithm used in the hdiv
+        # intrinisc. A smaple of values were extracted from the Nvidia hdiv
+        # test suite
+        arry1 = np.array([0x5b, 0xaf, 0xcb, 0x13b, 0x157, 0x173, 0x18f,
+                          0x1ab, 0x26f, 0x28b, 0x2a7, 0x2c3,
+                          0x2df, 0x2fb, 0x317, 0x333, 0x34f, 0x36b, 0x4b,
+                          0xa9, 0xb6, 0xc3, 0x13b, 0x15e, 0x177, 0x179,
+                          0x196, 0x1b3, 0x22b, 0x267, 0x276]).astype(np.float16)
+
+        arry2 = np.array([0x4b00, 0x4b00, 0x4b00, 0x4b00, 0x4b00, 0x4b00,
+                          0x4b00, 0x4b00, 0x4b00, 0x4b00, 0x4b00, 0x4b00,
+                          0x4b00, 0x4b00, 0x4b00, 0x4b00, 0x4b00, 0x4b00,
+                          0x4f80, 0x4e80, 0x4f00, 0x4f80, 0x4f80, 0x4f00,
+                          0x4f80, 0x4e80, 0x4f00, 0x4f80, 0x4f80, 0x4f80,
+                          0x4f00]).astype(np.float16)
+
+        ary = np.zeros_like(arry1, dtype=np.float16)
+        compiled.forall(ary.size)(ary, arry1, arry2)
+        ref = arry1 / arry2
+        np.testing.assert_allclose(ary, ref)
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_hdiv_unique_reg_predicate(self):
+        args = (f2[:], f2, f2)
+        ptx, _ = compile_ptx(simple_hdiv_unique, args, cc=(5, 3))
+
+        # Look for two .reg .pred __$temp3 lines preceded by a '{'
+        # which disambiguates (via nesting) the variable names.
+
+        predicate_count = 0
+        for line in ptx.split('\n'):
+            if '{ .reg .pred __$temp3;' in line:
+                predicate_count += 1
```

Comment:
Can be simplified to:
```suggestion
        predicate_count = ptx.count('{ .reg .pred __$temp3;')
```